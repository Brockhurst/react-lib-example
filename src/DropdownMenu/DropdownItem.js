import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export class DropdownItem extends React.Component {
  className = 'vg-dropdown-menu__item';

  getContainerClassName = () => {
    return classNames(this.className, { [`${this.className}__disabled`]: this.props.disabled });
  };

  onClick = () => {
    const { disabled, onClick, toggle } = this.props;

    if (!disabled) {
      onClick();
      toggle();
    }
  };

  render() {
    const { disabled, children } = this.props;

    return (
      <button
        type="button"
        className={this.getContainerClassName()}
        onClick={disabled ? undefined : this.onClick}
        disabled={disabled}
      >
        {children}
      </button>
    );
  }

  static propTypes = {
    onClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
  };

  static defaultProps = {
    disabled: false,
  };
}
