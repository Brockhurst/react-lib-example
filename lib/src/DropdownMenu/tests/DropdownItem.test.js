import React from 'react';
import { shallow } from 'enzyme';

import { DropdownItem } from '../DropdownItem';

describe('Dropdown Item', () => {
  let onClick;
  let toggle;

  const setup = (props = {}) => shallow(
    <DropdownItem onClick={onClick} toggle={toggle} {...props}>
      Dropdown Item
    </DropdownItem>,
  );

  beforeEach(() => {
    onClick = jest.fn();
    toggle = jest.fn();
  });

  it('should render correctly', () => {
    const notDisabledComponent = setup({ disabled: false });
    const disabledComponent = setup({ disabled: true });

    expect(notDisabledComponent).toMatchSnapshot();
    expect(disabledComponent).toMatchSnapshot();
  });

  it('should call toggle and onClick when it is clicked', () => {
    const component = setup();

    component.simulate('click');

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(toggle).toHaveBeenCalledTimes(1);
  });

  it('should not call toggle and onClick when it is clicked and disabled', () => {
    const component = setup({ disabled: true });

    component.simulate('click');

    expect(onClick).toHaveBeenCalledTimes(0);
    expect(toggle).toHaveBeenCalledTimes(0);
  });
});
