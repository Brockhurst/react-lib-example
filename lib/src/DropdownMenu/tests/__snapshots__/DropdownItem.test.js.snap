// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Item should render correctly 1`] = `
<button
  className="vg-dropdown-menu__item"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Dropdown Item
</button>
`;

exports[`Dropdown Item should render correctly 2`] = `
<button
  className="vg-dropdown-menu__item vg-dropdown-menu__item__disabled"
  disabled={true}
  type="button"
>
  Dropdown Item
</button>
`;
